docs/docs/adding-react-components.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Adding React Components
+---
+
+This guide covers how to add React components to your Gatsby site.
+
+## React components
+
+React components are prebuilt elements or groups of elements that can be used to prevent repeating yourself in your code.
+
+Components also offer the ability to be customised using "props" (properties). Props can be of any Javascript type, such as boolean, string, object, array or almost anything you can think of.
+
+For example, you could use a component for Buttons on your site. This is because they would likely be used multiple times across pages and would need different labels each time.
+
+## Importing React components
+
+In Gatsby, when using React components, you import and use them just like you would in normal React:
+
+```jsx
+import React from "react"
+import { Link } from "gatsby"
+
+export default () => (
+  <div>
+    <Link to="/contact/">Contact</Link>
+  </div>
+)
+```
+
+## Things to watch out for
+
+Since Gatsby uses Server Side Rendering (SSR) to generate your site's pages, the JSX code you write is usually compiled before the browser loads the page. Because of this, certain features are not available at compile time and can cause a build to error.
+
+### Use of browser globals
+
+Some components or code references on "browser globals" such as `window`, `document` or `localStorage`. This object is not available at compile time so can result in a Webpack error when building:
+
+```
+WebpackError: ReferenceError: window is not defined
+```
+
+#### Fixing your code
```

Comment:
Similar note to section below on line 93, since this content lives in the Porting from CRA to Gatsby guide, and the examples are mostly relevant to Debugging HTML Builds, how about we add these examples to the HTML Builds guide, and then cross reference it here (and then remove this section, opting for shorter notes with links instead)?

The line above with the [WebpackError](https://github.com/gatsbyjs/gatsby/pull/18651/files#diff-6983458a577ee39b5741ef3ec632559fR39) could link to the Debugging HTML Builds page which serves as the standard for solving issues in regards to builds, and this page can focus mainly on adding react components, and link off to those places with warnings about things to be aware of.